src/psyclone/transformations.py

```diff
@@ -340,67 +379,124 @@ def name(self):
         ''' Returns the name of this transformation as a string.'''
         return "DynamoLoopFuse"
 
-    def apply(self, node1, node2, same_space=False):
-        '''
-        Fuse the two Dynamo loops represented by :py:obj:`node1` and
-        :py:obj:`node2`. The optional same_space flag asserts that an
-        unknown iteration space (i.e. any_space) matches the other
-        iteration space. This is set at the users own risk.
+    def _validate(self, node1, node2):
```

Comment:
It is different to `apply()` having different number of arguments because `apply()` is not declared as a private method (with an underscore `_`) as is `_validate`. Apparently, having different number of arguments  in children classes' methods works for public methods but not for private. My original intention was to pass `same_space` as an argument to the `DynamoLoopFuseTrans._validate`  method because I assumed it would be possible to "overload" the functionality (different number of arguments) of the parent class. However, this failed miserably because the parent `LoopFuseTrans.apply` method complained about the positional arguments when it called `self._validate(node1, node2)`. Making `validate` a public method removed this issue but it is in disagreement with how `_validate` is applied in the rest of transformations.

Reply:
Used setter as suggested in the previous comment and added TODO with the reference to issue #478, as having this functionality will make the setter redundant.